realm/realm-library/src/main/java/io/realm/internal/Table.java

```diff
@@ -399,11 +420,12 @@ public long addEmptyRowWithPrimaryKey(Object primaryKeyValue) {
             switch (type) {
                 case STRING:
                 case INTEGER:
-                    if (findFirstNull(primaryKeyColumnIndex) != NO_MATCH) {
+                    if (!skipChecking && findFirstNull(primaryKeyColumnIndex) != NO_MATCH) {
                         throwDuplicatePrimaryKeyException("null");
                     }
                     rowIndex = nativeAddEmptyRow(nativePtr, 1);
                     row = getUncheckedRow(rowIndex);
+                    // FIXME: Use core's set_null_unique when core supports it.
```

Comment:
No, we still support `null` for PK. Just not calling the right method from core. But for now, it is not a problem at all since we guarantee the uniqueness of PK in binding level